include/frozen/map.h

```diff
@@ -27,52 +27,54 @@
 #include "frozen/bits/basic_types.h"
 #include "frozen/bits/constexpr_assert.h"
 #include "frozen/bits/exceptions.h"
+#include "frozen/bits/mpl.h"
 #include "frozen/bits/version.h"
 
+#include <iterator>
 #include <utility>
 
 namespace frozen {
 
 namespace impl {
 
-template <class Comparator> class CompareKey {
-
-  Comparator const comparator_;
-
+template <class Comparator> class CompareKey : private Comparator {
 public:
+  constexpr Comparator const& key_comp() const noexcept {
```

Comment:
I'm not sold on the benefits of this approach over membership. But I'm ready to be convinced!

Reply:
This is the same reason as I described in https://github.com/serge-sans-paille/frozen/pull/158#discussion_r1276040582 for `Hasher`: it is the (admittedly ugly) way to use the empty base optimization.